src/solve.jl

```diff
@@ -122,19 +122,46 @@ function solve(system::InitialValueProblem,
         transformation_matrix = nothing
     end
 
-    # ==============================
-    # Sparse/dense matrix conversion
-    # ==============================
+    # ===================================
+    # Sparse/dense/lazy matrix conversion
+    # ===================================
     A = Δ.s.A
+    create_new_system = false
+    if !options[:lazy_expm] && options[:lazy_expm_discretize]
```

Comment:
you can probably just keep `if !options[:lazy_expm]`

Reply:
No, otherwise the matrix may not be lazy :wink:

Reply:
but then check if the matrix if it is lazy (a `SparseMatrixExp`)

Reply:
too late :snail: